server/pulp/server/webservices/views/tasks.py

```diff
@@ -17,6 +21,10 @@
                                                 generate_json_response_with_pulp_encoder)
 
 
+# This constant set is used for deleting the completed tasks from the collection.
+VALID_STATES = set(filter(lambda state: state != CALL_CANCELED_STATE, CALL_COMPLETE_STATES))
```

Comment:
As an alternative, the states can be defined here instead of importing from constants.